aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -87,6 +87,16 @@ Tensor isnan(const Tensor& self) {
   return self != self;
 }
 
+Tensor isfinite(const Tensor& self) {
+  // Integral tensor types are finite
+  if (!self.is_floating_point()) {
+    return at::ones_like(self, at::kBool, at::MemoryFormat::Preserve);
+  }
+  return AT_DISPATCH_FLOATING_TYPES_AND_HALF(self.scalar_type(), "isfinite", [&]() {
+    return (self == self) * (self.abs() != at::full(1, std::numeric_limits<scalar_t>::infinity(), self.options()));
```

Comment:
Does `self.abs() != std::numeric_limits<double>::infinity()` work? I'm not sure how the `Scalar` API works or if matters at all.

Reply:
Looks like it works